File: Graph/stack.java
public class stack {
    private int top;
    private int[] arr;

    public stack(int size) {
        arr = new int[size];
        top = -1;
    }

    public void push(int data) {
        if (top == arr.length - 1) {
            System.out.println("Stack overflow");
            return;
        }
        arr[++top] = data;
    }

    public int pop() {
        if (top == -1) {
            System.out.println("Stack underflow");
            return -1;
        }
        return arr[top--];
    }

    public int peek() {
        if (top == -1) {
            System.out.println("Stack is empty");
            return -1;
        }
        return arr[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public int size() {
        return top + 1;
    }

    public static void main(String[] args) {
        stack stack = new stack(5);
        stack.push(1);
        stack.push(2);
        stack.push(3);
        stack.push(4);
        stack.push(5);

        System.out.println("Peek: " + stack.peek()); // Peek: 5
        System.out.println("Pop: " + stack.pop()); // Pop: 5
        System.out.println("Size: " + stack.size()); // Size: 4
        System.out.println("Pop: " + stack.pop()); // Pop: 4
        System.out.println("Size: " + stack.size()); // Size: 3
        System.out.println("Peek: " + stack.peek()); // Peek: 3
        System.out.println("Pop: " + stack.pop()); // Pop: 3
        System.out.println("Size: " + stack.size()); // Size: 2
        System.out.println("Pop: " + stack.pop()); // Pop: 2
        System.out.println("Size: " + stack.size()); // Size: 1
        System.out.println("Pop: " + stack.pop()); // Pop: 1
        System.out.println("Size: " + stack.size()); // Size: 0
        System.out.println("Pop: " + stack.pop()); // Stack underflow
    }
}

File: DSA class/stack.cpp
//implementation of stack?

#include <iostream>

using namespace std;

// Define the stack data structure.
struct Stack {
  int data;
  Stack *next;
};

// Create a new stack.
Stack *createStack() {
  Stack *newStack = new Stack;
  newStack->data = 0;
  newStack->next = NULL;

  return newStack;
}

// Push an item onto the stack.
void push(Stack *stack, int item) {
  Stack *newNode = new Stack;
  newNode->data = item;
  newNode->next = stack->next;

  stack->next = newNode;
}

// Pop an item off the stack.
int pop(Stack *stack) {
  int item = stack->next->data;
  Stack *temp = stack->next;

  stack->next = stack->next->next;

  delete temp;

  return item;
}

// Check if the stack is empty.
bool isEmpty(Stack *stack) {
  if (stack->next == NULL) {
    return true;
  } else {
    return false;
  }
}

// Print the contents of the stack.
void printStack(Stack *stack) {
  Stack *current = stack->next;

  while (current != NULL) {
    cout << current->data << " ";
    current = current->next;
  }

  cout << endl;
}

int main() {
  // Create a new stack.
  Stack *stack = createStack();

  // Push some items onto the stack.
  push(stack, 10);
  push(stack, 20);
  push(stack, 30);

  // Print the contents of the stack.
  printStack(stack);

  // Pop an item off the stack.
  int item = pop(stack);

  // Print the item that was popped.
  cout << "The item that was popped is: " << item << endl;

  // Check if the stack is empty.
  if (isEmpty(stack)){
    cout<<"satck is empty";
  }
  else{
    cout<<"Stack is not empty";
  }

}